app/services/setting_service.rb

```diff
@@ -0,0 +1,70 @@
+require_relative "data_sources/json_setting_data_source"
+require_relative "../shared/logging_module"
+require_relative "../shared/available_settings"
+require_relative "./services"
+
+module FastlaneCI
+  class SettingServiceKeyNotFoundError < FastlaneCIError
+  end
+
+  # Provides access to system settings
+  class SettingService
+    include FastlaneCI::Logging
+    attr_accessor :setting_data_source
+
+    def initialize(setting_data_source: nil)
+      unless setting_data_source.nil?
+        unless setting_data_source.class <= SettingDataSource
+          raise "setting_data_source must be descendant of #{SettingDataSource.name}"
+        end
+      end
+
+      if setting_data_source.nil?
+        # Default to JSONSettingDataSource
+        # TODO: do we need `sample_data` here?
+        logger.debug(
+          "setting_data_source is new, using `ENV[\"data_store_folder\"]` " \
+          "if available, or `sample_data` folder"
+        )
+        data_store_folder = ENV["data_store_folder"] # you can set it at runtime!
+        data_store_folder ||= File.join(FastlaneCI::FastlaneApp.settings.root, "sample_data")
+        setting_data_source = JSONSettingDataSource.create(data_store_folder)
+      end
+
+      self.setting_data_source = setting_data_source
```

Comment:
I think this should be named `settings_data_source` instead of the singular: `setting_data_source`

Reply:
Nice catch, will update 👍

Reply:
Actually no, seems like we're using the singular everywhere, including https://github.com/fastlane/ci/blob/master/app/services/build_service.rb#L16 and https://github.com/fastlane/ci/blob/master/app/services/environment_variable_service.rb#L30